GUIDELINES.md

```diff
@@ -164,3 +164,36 @@ To support this use case, you can choose one of the following options:
     };
     ...
     ```
+
+## Diag Channel
+
+The diag channel can be used to troubleshoot issues with instrumentation packages.
+
+### Patching Messages
+
+When OpenTelemetry is installed in a user application, and expected spans are missing from generated traces, it is often useful to differentiate between the following scenarios:
+
+- The instrumentation is not auto loaded - due to issue with the require/import interception, an unsupported version of the instrumented package, or some other issue. This knowledge can pin-point the issue to the instrumentation package.
+- The instrumentation patch was applied but expected spans are missing - this can spot the light on the instrumented package logic, configuration, limits, otel sdk, or other issues.
+
+It can also be useful to know when the instrumentation is loaded and patched, to understand the order of operations in the application.
+
+Instrumentation packages should use the `@opentelemetry/instrumentation` package `BaseInstrumentation` class to register patches and unpatch callbacks for specific require/import of the instrumented package, it's dependency or an internal module file. When this mechanism is used, the base class will automatically emit a debug message on instrumentation diag component logger, looking like this:
+
+```shell
+@opentelemetry/instrumentation-foo Applying instrumentation patch for module on require hook {
+  module: 'foo',
+  version: '1.2.3',
+  baseDir: '<your directory>/node_modules/mongoose'
+}
+```
+
+Instrumentation should not add additional debug messages for triggering the patching and unpatching callbacks, as the base class will handle this.
+
+Instrumentation may add additional patch/unpatch messages for specific functions if it is expected to help in troubleshooting issues with the instrumentation. Few examples:
```

Comment:
Oh, another thought I had that I think will probably be updated in the follow-up PR that replaces inconsistent usages of `api.diag.debug` and similar that aren't using the instrumentation `this` to allow for `this._diag.debug`. When we make that change we should update guidelines for that as well, though it will already be easier to follow suit once these are all more consistent 😄

Reply:
certainly, I aim to first remove or hoist unneeded diag messages, and then will transform all remaining diag message to use `this._diag` and update the GUIDELINES.md 👍 

https://github.com/open-telemetry/opentelemetry-js/pull/4663 will also help in hosting many diag prints and remove the repeated code from  dozens of places in contrib.